linodecli/plugins/ssh.py

```diff
@@ -0,0 +1,107 @@
+"""
+The ssh plugin allows sshing into Linodes by label or ID
+
+Invoke as follows::
+
+   linode-cli ssh [USERNAME@]LINODE_LABEL [SSH ARGS..]
+
+   LINODE_LABEL - the label of the Linode to ssh into
+   USERNAME - the user to ssh into the Linode as.  Defaults to the current user
+"""
+import argparse
+import subprocess
+from sys import exit, platform, version_info
+
+
+def input_helper(prompt):
+    """
+    Handles python2 and python3 differences in input command
+    """
+    if version_info[0] == 2:
+        # python2 input is scary - we want raw_input
+        return raw_input(prompt)
+    else:
+        return input(prompt)
+
+
+def call(args, context):
+    """
+    Invokes this plugin
+    """
+    if platform == 'win32':
+        print('This plugin is not currently supported in Windows.  For more '
+              'information or to suggest a fix, please visit '
+              'https://github.com/linode/linode-cli')
+        exit(1)
+
+
+    parser = argparse.ArgumentParser("linode-cli ssh", add_help=True)
+    parser.add_argument('label', metavar='[USERNAME@]LABEL', nargs='?', type=str,
+                        help="The label of the Linode to SSH into, optionally with "
+                             "a username before it in USERNAME@LABEL format.  If no "
+                             "username is given, defaults to the current user.")
+
+    parsed, args = parser.parse_known_args(args)
+
+    if not parsed.label:
+        parser.print_help()
+        exit(0)
+
+    label = parsed.label
+    username = None
+    if '@' in parsed.label:
+        username, label = parsed.label.split('@', 1)
+
+    result, potential_matches = context.client.call_operation(
+            "linodes", "list", filters={'label': {"+contains": label}})
+
+    if result != 200:
+        # TODO
+        print('Something went wrong')
+        exit(2)
+
+    potential_matches = potential_matches['data']
+    exact_match = None
+
+    # see if we got a match
+    for match in potential_matches:
+        if match['label'] == label:
+            exact_match = match
+            break
+
+    if exact_match is None:
+        # no match - stop
+        print("No Linode found for label {}".format(label))
+
+        if potential_matches:
+            print('Did you mean: ')
+            print('\n'.join([' {}'.format(p['label']) for p in potential_matches]))
+        exit(1)
+
+    if exact_match['status'] != 'running':
+        print('{} is not running (status is {}); operation aborted.'.format(
+            label, exact_match['status']))
+        exit(2)
+
+    # find a public IP Address to use
+    public_ip = None
+    for ip in exact_match['ipv4']:
```

Comment:
A `-6` option would make sense to permit users to get the public IPv6 address instead.

`-6` is convention in many tools, including ssh.

Reply:
Linodes come with a `/64` for slaac addresses - what of those addresses should we select to SSH to?  The lowest?

Reply:
Users only get the one address described as the SLAAC address.  The `/64` in this case is not addressable by the user, it is their network of accessible peers that don't require a gateway (`fe80::1`) to reach.  

This is a confusing feature of the API because the IPv4 addresses do not return `/24` and `/17` (for private addresses) which would be inline with the use of `/64` in the SLAAC response. 

In the case of IPv6 Ranges, the entire `/116` is user addressable (across all nodes in the region) and any address outside of that range must use the `fe80::1` gateway to be reached.